lib/kubernetes-deploy/container_logs.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class ContainerLogs
+    attr_reader :lines, :container_name
+
+    DEFAULT_LINE_LIMIT = 250
+
+    def initialize(parent_id:, container_name:, logger:)
+      @parent_id = parent_id
+      @container_name = container_name
+      @logger = logger
+      @lines = []
+      @last_printed_index = -1
```

Comment:
Nitpick: If we call this something like `@next_print_index`, then you might be able to simplify things a tiny bit. This could be initialized to `0` here and below the `+1` and`-1` logic would be unnecessary, you could just do `lines[@next_print_index .. -1]` and avoid the extra `start_at` variable.